gridly/src/grid/bounds.rs

```diff
@@ -312,6 +312,22 @@ impl Display for BoundsError {
     }
 }
 
+#[cfg(feature = "array_impls")]
+impl<T, const N: usize, const M: usize> GridBounds for [[T; M]; N] {
+    #[inline]
+    fn dimensions(&self) -> Vector {
+        assert!(M <= core::isize::MAX as usize);
```

Comment:
It should be possible to use `static_assertions` here

Reply:
I don't believe it's possible (yet) - `const_assert` doesn't work with const generics (https://github.com/nvzqz/static-assertions-rs/issues/40), and the workaround are is not ideal; it's possible to use some associated const hacks to have it checked in compile time, see https://github.com/nvzqz/static-assertions-rs/issues/40#issuecomment-846228355, however the error that's shown is very non indicative since it doesn't know where the array type is "instantiated", it will show an error like:

```
error[E0080]: evaluation of `gridly::grids::array::<impl gridly::grid::GridBounds for [[T; C]; R]>::dimensions::Assert::<usize::MAX>::OK` failed
  --> /Users/jon/foo/gridly-rs/gridly/src/grids/array.rs:23:9
   |
23 |         const_assert!(R: usize => R <= core::isize::MAX as usize); // Row count overflow check
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_u8 - 1_u8`, which would overflow
   |
   = note: this error originates in the macro `const_assert` (in Nightly builds, run with -Z macro-backtrace for more info)
```

which isn't very useful for library code IMO